src/contrib/cluster/Akka.DistributedData/Serialization/ReplicatedDataSerializer.cs

```diff
@@ -251,57 +251,74 @@ private static Type GetTypeFromDescriptor(TypeDescriptor t)
         }
 
         #region ORSet
-
-        private static Proto.Msg.ORSet ORSetToProto<T>(ORSet<T> set)
-        {
-            var p = new Proto.Msg.ORSet();
-            p.Vvector = SerializationSupport.VersionVectorToProto(set.VersionVector);
-            p.Dots.Add(set.ElementsMap.Values.Select(SerializationSupport.VersionVectorToProto));
-            p.TypeInfo = new TypeDescriptor();
-            return p;
-        }
         private IORSet ORSetFromBinary(byte[] bytes)
         {
             return FromProto(Proto.Msg.ORSet.Parser.ParseFrom(bytes));
         }
 
         private Proto.Msg.ORSet ToProto(IORSet orset)
         {
+            var b = new Proto.Msg.ORSet
+            {
+                TypeInfo = new TypeDescriptor()
+            };
+
             switch (orset)
             {
                 case ORSet<int> ints:
                     {
-                        var p = ORSetToProto(ints);
-                        p.TypeInfo.Type = ValType.Int;
-                        p.IntElements.Add(ints.Elements);
-                        return p;
+                        b.Vvector =  SerializationSupport.VersionVectorToProto(ints.VersionVector);
+                        b.TypeInfo.Type = ValType.Int;
+                        var intElements = new List<int>(ints.ElementsMap.Keys);
+                        intElements.Sort();
+                        foreach (var val in intElements)
+                        {
+                            b.Dots.Add(SerializationSupport.VersionVectorToProto(ints.ElementsMap[val]));
```

Comment:
Version vectors needs to be re-arranged so that it matches the newly sorted keys, so that it can be correctly deserialized into dictionary during deserialization.